plugins/modules/sda_host_port_onboarding_workflow_manager.py

```diff
@@ -0,0 +1,3010 @@
+            This method calls the SDA 'get_fabric_sites' API to retrieve the fabric ID for a specified site. It logs the response,
+            processes the response to extract the fabric ID, and handles any exceptions that occur during the API call.
+        """
+        try:
+            # Call the SDA 'get_fabric_sites' API with the provided site ID
+            response = self.dnac._exec(
+                family="sda",
+                function="get_fabric_sites",
+                op_modifies=True,
+                params={"siteId": site_id},
+            )
+            self.log("Response received post SDA - 'get_fabric_sites' API call: {0}".format(str(response)), "DEBUG")
+
+            response = response.get("response")
+            if not response:
+                self.log("No response received from the SDA - 'get_fabric_sites' API call.", "WARNING")
+                return None
+
+            fabric_id = response[0]["id"]
+            return fabric_id
+
+        except Exception as e:
+            # Log an error message and fail if an exception occurs
+            self.msg = (
+                "An error occurred while retrieving fabric Site 'Id' for Site '{0}' using SDA - "
+                "'get_fabric_sites' API call: {1}".format(site_name, str(e))
+            )
+            self.fail_and_exit(self.msg)
+
+    def get_network_fabric_id(self, ip_address, hostname):
+        """
+        Retrieves the network fabric ID for a device using its IP address or hostname.
+        Args:
+            ip_address (str): The management IP address of the device.
+            hostname (str): The hostname of the device.
+        Returns:
+            tuple: A tuple containing two values:
+                - mgmt_ip_to_instance_id_map (dict): A dictionary mapping management IP addresses to device instance IDs.
+                - fabric_id (str): The ID of the fabric to which the device belongs.
+        Description:
+            This method retrieves the network fabric ID for a device by first getting the device's instance ID using its
+            IP address or hostname. It then retrieves the site name and site ID where the device is located, and finally
+            obtains the fabric ID of the site. The method logs relevant information and returns the instance ID map and
+            the fabric ID.
+        """
+        get_device_list_params = self.get_device_list_params(ip_address, hostname)
+
+        # Get Device IP Address and Id (networkDeviceId required)
+        mgmt_ip_to_instance_id_map = self.get_device_ids_by_params(get_device_list_params)
+        self.log("Collected mgmt_ip_to_instance_id_map: {0}".format(mgmt_ip_to_instance_id_map), "DEBUG")
+
+        # Get the Site Name the device is part of.
+        site_name = self.get_device_info_from_sda_fabric(list(mgmt_ip_to_instance_id_map.keys())[0])
+
+        # Get siteId of the Site the device is part of
+        site_exists, site_id = self.get_site_id(site_name)
+
```

Comment:
if not site_exists:
        self.msg = "Site ID not found for site name: {0}".format(site_name)
        Either print error and return or fail json
        self.fail_and_exit(self.msg)